modules/gdscript/gdscript_parser.cpp

```diff
@@ -4087,8 +4089,15 @@ String GDScriptParser::DataType::to_string() const {
 			}
 			return native_type.operator String();
 		}
-		case ENUM:
-			return enum_type.operator String() + " (enum)";
+		case ENUM: {
+			if (native_type != StringName()) {
+				// If native type is a file, hide file path and keep only file name
```

Comment:
I'm wondering. If my file name is `A.gd` and I import it by doing `const B = preload("A.gd")`, will it write `A` as the file type?

Reply:
This is just for enums, so I'm not sure! So you mean preloading an external class that has an enum? Probably should add a test for that.

Something like:

```GDScript
const B = preload("A.gd")
make_error_with(B.SomeEnum)
```

Reply:
Adding the following test:

```GDScript
enum LocalNamed { VALUE_A, VALUE_B, VALUE_C = 42 }

func test():
	const P = preload("../features/enum_from_outer.gd")
	var x : LocalNamed
	x = P.Named.VALUE_A
```

Output:

```GDTEST_ANALYZER_ERROR
Value of type "enum_from_outer.gd::Named" cannot be assigned to a variable of type "preload_enum_error.gd::LocalNamed".
```

Reply:
I mean this:
```gdscript
# b.gd
enum {
    VALUE_A,
    VALUE_B,
    VALUE_C
}
```
```gdscript
# test.gd
enum LocalNamed {
    VALUE_A,
    VALUE_B,
    VALUE_C
}

func test():
    const A = preload("b.gd")
    var x: LocalNamed
    x = A.VALUE_A
```

Reply:
oh, interesting, I'm curious to see the error message even without the `preload`! Not sure how named & unnamed enums interact.

Reply:
I'd forgotten that I had accounted for this :)

```GDTEST_ANALYZER_ERROR
Value of type "enum_value_from_parent.gd::UNNAMED_ENUM" cannot be assigned to a variable of type "enum_preload_unnamed_assign_to_named.gd::MyEnum".
```

Added tests for this :)

Reply:
I wonder if it would be better to name `UNNAMED_ENUM` something like `<enum>`, so it reads: `b.gd::<enum>`.

I have the impression that otherwise, it could conflict with
```gdscript
# enum_value_from_parent.gd
enum UNNAMED_ENUM {
    IM_SO_EDGY,
    WITH_MY_IRONIC_NAME
}
```

Reply:
Also, I think it should use the entire resource path for the reporting, as there could be multiple `enum_value_from_parent.gd` files in a project.

Reply:
I'll change it to something like `<unnamed enum>` - I would like to make it explicit that it's an enum without names, not just any enum, but you're right, the `<>` is a good indicator!

The issue with having the path is that the path is absolute, so if I generate tests locally, they won't pass on github. Perhaps I could find a way to change the path to a relative path though. 

But people would need to define two .gd files with the same name but different paths, both with unnamed enums. And then get their enums confused? I feel like that's enough of an edge case that maybe it's not worth the longer/more confusing error message in most cases we expect. What do you think?

Reply:
> The issue with having the path is that the path is absolute, so if I generate tests locally, they won't pass on github. Perhaps I could find a way to change the path to a relative path though.

Why not the `res://` path?

Reply:
The datatype are constructed using the class' fully qualified class name. I'm not sure we do have access to the `res://` path in the analyzer, though it'd be neat if we did?